fix(#7007): Implements Vault Health Check integration.

lordofthejars opened the pull request:
Fixes #7007

loicmathieu committed:
Provides multi-tenancy capabilities for MongoDB with Panache

oztimpower committed:
Deployment changes for BuildSteps with RequestStreamHandler, and Runtime with AmazonLambdaRecorder

patriot1burke committed:
Quarkus Funqy FaaS framework initial commit

graal workaround

lambda

fix pom

rebase and stream test

stuartwdouglas committed:
Attempt to fix intermittent failure

mariofusco committed:
remove kogito extension from quarkus codebase

manovotn committed:
Support for repeating qualifiers in ArC plus automated tests.

geoand committed:
Fix Optaplanner integration test

mariofusco committed:
remove kogito from CI

mkouba committed:
ArC - transform unproxyable bean classes by default

- new transformations: remove final flag from a bean class, change
visibility of private no-args constructor
- deprecate quarkus.arc.remove-final-for-proxyable-methods
- introduce quarkus.arc.transfrorm-unproxyable-classes
- this should also help kotlin developers where classes are final by
default

rsvoboda committed:
Close local KubernetesClient

mkouba committed:
MP Config TCK - disable transformation of unproxyable types

gastaldi committed:
Create config.yml for GitHub issues

Co-Authored-By: Guillaume Smet <guillaume.smet@gmail.com>

stuartwdouglas committed:
Allow configuration of max frame size

Fixes #7172

geoand committed:
Use proper configuration for environment variables in JibConfig

gastaldi committed:
Avoid null check

Config elements are never null

tobiasstadler committed:
Updated Jackson to version 2.10.3

Postremus committed:
Have the Agroal datasources initialized eagerly

stuartwdouglas committed:
Make mockito parent first

This is a stop gap measure, we need
a more extensible solution to this.

Fixes #7854

gdoenlen committed:
Add documentation for building native on windows

Now that compiling to native on windows is working properly add
documentation on the steps required as discussed in #5675

lordofthejars committed:
fix(#7865): Adds support for Vault TOTP secrets

mkouba committed:
Make it possible to dump zig files in all modes

- resolves #7844

mkouba committed:
Update BeanProcessor.Builder.setTransformUnproxyableClasses() javadoc

jmartisk committed:
Make use of ExtendedMetadataBuilder

geoand committed:
Make @QuarkusTest play nicely with @ParameterizedTest

Fixes: #7877

dependabot-preview[bot] committed:
Bump kubernetes-client-bom from 4.8.0 to 4.9.0

Bumps [kubernetes-client-bom](https://github.com/fabric8io/kubernetes-client) from 4.8.0 to 4.9.0.
- [Release notes](https://github.com/fabric8io/kubernetes-client/releases)
- [Changelog](https://github.com/fabric8io/kubernetes-client/blob/master/CHANGELOG.md)
- [Commits](https://github.com/fabric8io/kubernetes-client/compare/v4.8.0...v4.9.0)

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

geoand committed:
Add tests to verify mockito is working

mkouba committed:
Update independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/ComponentsProviderGenerator.java

Co-Authored-By: George Gastaldi <gegastaldi@gmail.com>

mkouba committed:
Undeploy verticles backed by contextual instances

...of ApplicationScoped beans before the application context is destroyed

Update extensions/vertx/deployment/src/test/java/io/quarkus/vertx/verticles/NotDeployedVerticle.java

Co-Authored-By: George Gastaldi <gegastaldi@gmail.com>

Update extensions/vertx/deployment/src/test/java/io/quarkus/vertx/verticles/MyUndeployedVerticle.java

Co-Authored-By: George Gastaldi <gegastaldi@gmail.com>

sdaschner committed:
change remote debug connection string for JDK 9+
fixes #7889

lordofthejars committed:
fix(#7007): Implements Vault Health Check integration.

sberyozkin commented on extensions/vault/runtime/src/main/java/io/quarkus/vault/health/VaultHealthCheck.java:
```diff
@@ -0,0 +1,57 @@
+        final HealthCheckResponseBuilder builder = HealthCheckResponse.named("Vault connection health check");
+
+        try {
+            final VaultHealth vaultHealth = this.vaultSystemBackendEngine.health(false, false);
```
The standby and perf customizations are not possible here

lordofthejars replied:
Not here, but if they are configured in `application.properties`, then these values are overridden. I know that you are asking why are you leaving this instead of provide a `health()` method. The reason is that `VaultSystemBackendEngine` could be injected in any class and users programmatically might want to set them programmatically.

sberyozkin replied:
@lordofthejars thanks, but then we'd support the users overwriting the configuration, and when the config is good, then they'd have to type `false, false` :-)
Having `health()` alone is quite enough IMHO, it does seem to me in most cases health of the running active Vault would be of interest. `health(boolean standbyok)` and `health(boolean perfok)` can be easily added if someone ever asks :-)

sberyozkin commented on extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/VaultSystemBackendManager.java:
```diff
@@ -0,0 +1,32 @@
+    }
+
+    @Override
+    public VaultHealth health(boolean isStandByOk, boolean isPerfStandByOk) {
```
Would you like to keep these parameters to let the user code override the configuration properties ? I propose just drop them since the config properties win anyway and the user code does not probably need to know about these minor status tweaks

lordofthejars replied:
I copy the same comment before just to keep a relationship:

I know that you are asking why are you leaving this instead of provide a health() method. The reason is that VaultSystemBackendEngine could be injected in any class and users programmatically might want to set them programmatically.

sberyozkin commented on extensions/vault/runtime/src/main/java/io/quarkus/vault/runtime/config/VaultRuntimeConfig.java:
```diff
@@ -50,6 +51,28 @@
     @ConfigDocSection
     public VaultAuthenticationConfig authentication;
 
+    /**
```
I propose to introduce a `Health` configuration group as these 3 parameters really stand up on their own IMHO.
Also, is default value needed for `false` ?

lordofthejars replied:
Now, correct me if I am not right these properties are under: `quarkus.vault.health.*` isn't it?

Yes by default they are `false`

lordofthejars replied:
What I have seen in other extensions that they are integrating with the spec, they are following the same approach, but please I am new on extension ecosystem, so I can be totally wrong.

sberyozkin replied:
You'd have 
```
@ConfigItem
Health health = new Health();
@ConfigGroup
public class Health {
   boolean enabled;
   // etc
}
```

Sure lets keep a default initialization to `false`

sberyozkin replied:
@lordofthejars It is not a big deal, but it feels it would better be encapsulated in the dedicate group

sberyozkin commented:
@lordofthejars Hi, it looks like a very useful addition, lets see what Vincent @vsevel says

sberyozkin commented:
@lordofthejars or may be have 2 health() methods from the beginning :-), a no arg method to make a mainstream case easier to support, and with 2 booleans for the customizations

michalszynkiewicz committed:
exclude checker-qual

sdaschner committed:
change remote debug connection string for JDK 9+ for Gradle

dependabot-preview[bot] committed:
Bump jgit.version from 5.7.0.202003090808-r to 5.7.0.202003110725-r

Bumps `jgit.version` from 5.7.0.202003090808-r to 5.7.0.202003110725-r.

Updates `org.eclipse.jgit` from 5.7.0.202003090808-r to 5.7.0.202003110725-r

Updates `org.eclipse.jgit.http.server` from 5.7.0.202003090808-r to 5.7.0.202003110725-r

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

stuartwdouglas committed:
Resolve test jars from the local project correctly

Fixes #7887

stuartwdouglas committed:
Run QuarkusTestResource in the runtime ClassLoader

This is only for QuarkusTest, QuarkusUnitTest will need
a slightly different solution if we decide it is a problem, as
the lifecycles don't match.

Fixes #7880

lordofthejars commented:
@sberyozkin there some failures in CI/CD but running from Maven from my machine they are not happening.

The failure is:

```
2020-03-16T17:59:59.9522373Z [INFO] Running io.quarkus.vault.VaultITCase
2020-03-16T18:00:00.1382299Z [ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.177 s <<< FAILURE! - in io.quarkus.vault.VaultITCase
2020-03-16T18:00:00.1388461Z [ERROR] io.quarkus.vault.VaultITCase  Time elapsed: 0.177 s  <<< ERROR!
2020-03-16T18:00:00.1398329Z java.lang.RuntimeException: java.lang.RuntimeException: java.lang.RuntimeException: Failed to load steps from class io.quarkus.vault.VaultProcessor
2020-03-16T18:00:00.1404703Z 	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:427)
2020-03-16T18:00:00.1413271Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.lambda$invokeBeforeAllCallbacks$7(ClassBasedTestDescriptor.java:359)
2020-03-16T18:00:00.1421923Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-03-16T18:00:00.1429960Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.invokeBeforeAllCallbacks(ClassBasedTestDescriptor.java:359)
2020-03-16T18:00:00.1437115Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:189)
2020-03-16T18:00:00.1445676Z 	at org.junit.jupiter.engine.descriptor.ClassBasedTestDescriptor.before(ClassBasedTestDescriptor.java:78)
2020-03-16T18:00:00.1453922Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:132)
2020-03-16T18:00:00.1465664Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-03-16T18:00:00.1496795Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-03-16T18:00:00.1502335Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-03-16T18:00:00.1509810Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-03-16T18:00:00.1517231Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-03-16T18:00:00.1540396Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-03-16T18:00:00.1548070Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-03-16T18:00:00.1551781Z 	at java.util.ArrayList.forEach(ArrayList.java:1257)
2020-03-16T18:00:00.1562483Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
2020-03-16T18:00:00.1578429Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
2020-03-16T18:00:00.1585486Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-03-16T18:00:00.1600541Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
2020-03-16T18:00:00.1622399Z 	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
2020-03-16T18:00:00.1665052Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
2020-03-16T18:00:00.1672381Z 	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
2020-03-16T18:00:00.1679340Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
2020-03-16T18:00:00.1685341Z 	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
2020-03-16T18:00:00.1700172Z 	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
2020-03-16T18:00:00.1732895Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
2020-03-16T18:00:00.1741209Z 	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
2020-03-16T18:00:00.1747863Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:220)
2020-03-16T18:00:00.1774736Z 	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$6(DefaultLauncher.java:188)
2020-03-16T18:00:00.1781968Z 	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:202)
2020-03-16T18:00:00.1788204Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:181)
2020-03-16T18:00:00.1796125Z 	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:128)
2020-03-16T18:00:00.1797215Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:142)
2020-03-16T18:00:00.1798077Z 	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:117)
2020-03-16T18:00:00.1824594Z 	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
2020-03-16T18:00:00.1830485Z 	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
2020-03-16T18:00:00.1841250Z 	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
2020-03-16T18:00:00.1865547Z 	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
2020-03-16T18:00:00.1888285Z Caused by: java.lang.RuntimeException: java.lang.RuntimeException: Failed to load steps from class io.quarkus.vault.VaultProcessor
2020-03-16T18:00:00.1896800Z 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:206)
2020-03-16T18:00:00.1943324Z 	at io.quarkus.runner.bootstrap.AugmentActionImpl.createInitialRuntimeApplication(AugmentActionImpl.java:95)
2020-03-16T18:00:00.1948429Z 	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:386)
2020-03-16T18:00:00.1950047Z 	... 37 more
2020-03-16T18:00:00.1997638Z Caused by: java.lang.RuntimeException: Failed to load steps from class io.quarkus.vault.VaultProcessor
2020-03-16T18:00:00.1998611Z 	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:214)
2020-03-16T18:00:00.1999447Z 	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:136)
2020-03-16T18:00:00.2000273Z 	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:101)
2020-03-16T18:00:00.2001105Z 	at io.quarkus.runner.bootstrap.AugmentActionImpl.runAugment(AugmentActionImpl.java:204)
2020-03-16T18:00:00.2001941Z 	... 39 more
2020-03-16T18:00:00.2002826Z Caused by: java.lang.IllegalArgumentException: Run time configuration cannot be consumed here unless the method is a @Recorder at io.quarkus.vault.runtime.config.VaultRuntimeConfig arg0 of io.quarkus.smallrye.health.deployment.spi.HealthBuildItem io.quarkus.vault.VaultProcessor.addHealthCheck(io.quarkus.vault.runtime.config.VaultRuntimeConfig) of class io.quarkus.vault.VaultProcessor
2020-03-16T18:00:00.2003722Z 	at io.quarkus.deployment.ExtensionLoader.reportError(ExtensionLoader.java:992)
2020-03-16T18:00:00.2008270Z 	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:748)
2020-03-16T18:00:00.2013463Z 	at io.quarkus.deployment.ExtensionLoader.loadStepsFrom(ExtensionLoader.java:212)
2020-03-16T18:00:00.2015075Z 	... 42 more
```

Since I cannot reproduce locally and before I start doing pushes to test it, does it mean that I need to do:

```
@ConfigGroup
public class HealthConfig {

    public static final String DEFAULT_HEALTH_CHECK_ENABLED = "false";

    /**
     * Whether or not an health check is published in case the smallrye-health extension is present.
     */
    @Recorder
    @ConfigItem(name = "enabled", defaultValue = DEFAULT_HEALTH_CHECK_ENABLED)
    public boolean enabled;
}
```

geoand commented:
Hi @lordofthejars. 

What you are showing in the code snippet is not correct.
The error essentially tells you that:

```
    HealthBuildItem addHealthCheck(VaultRuntimeConfig config) {
        return new HealthBuildItem("io.quarkus.vault.health.VaultHealthCheck",
                config.healthEnabled, "vault");
    }
```

cannot consume `VaultRuntimeConfig`. This is because at build time (when `addHealthCheck` runs), the runtime value of `config` is not known.
The only legal operation that is permitted to do on a runtime config value, is to pass it to a recorder class (see https://quarkus.io/guides/writing-extensions#bytecode-recording for details on a  recorder).

I would suggest that you do a

```
./mvnw clean install -DskipTests -DskipITs -DskipDocs
```

and then run the tests locally. It is almost certain that you will be able to reproduce the problem you are seeing in CI.

lordofthejars commented:
@geoand So I need to annotate the Config class as `@Recorder`

geoand commented:
> @geoand So I need to annotate the Config class as `@Recorder`

That is not the correct use of `@Recorder`. See the extension author's guide I linked to

lordofthejars commented:
From Stuart conversation:

This is runtime config, but you are reading it at build time, so runtime changes would not be picked up.

You need to create a new config class that is build time config, see io.quarkus.vertx.http.runtime.HttpConfiguration and io.quarkus.vertx.http.runtime.HttpBuildTimeConfig for an example.

geoand commented:
> From Stuart conversation:
> 
> This is runtime config, but you are reading it at build time, so runtime changes would not be picked up.
> 
> You need to create a new config class that is build time config, see io.quarkus.vertx.http.runtime.HttpConfiguration and io.quarkus.vertx.http.runtime.HttpBuildTimeConfig for an example.

To expand on this: Build time configuration can be read without issues in build step. That is why Stuart proposes to split the vault configuration into build time and runtime (as other extensions do) and add the health configuration as a build time property.

lordofthejars commented:
I thought that this could happen when I was developing it but I was not super sure. Well, every day you learn something new.

geoand committed:
Fix intermittent Kubernetes test failures on Windows

lordofthejars commented:
@vsevel @geoand @stuartwdouglas The only failing it seems it is not related to my change as only fails in windows and in RestEasy module. The rest of the checks are successful.

sdaschner committed:
updated debug connection behavior for consistency

mkouba committed:
Make it possible to inject Instance<> into observers/disposers

- also destroy Instance<> correctly when the invocation completes
- resolves #7888

cescoffier committed:
update Mutiny to 0.4.2.

mkouba committed:
Refine BeanProcessor.Builder javadoc

- BeanProcessor.Builder.setRemoveFinalFromProxyableMethods()

Commit:
feat: add doc about defaults Spring Boot naming strategies and log about how to configure the Quarkus properties to use them

oztimpower committed:
Include Gradle support, JUnit tests, AWS SDK Java v2

jmartisk committed:
Fix display name of gc.total base metric

mkouba committed:
Qute type-safe templates - fix interfaces validation

- resolves #7843

mkouba committed:
Qute type-safe templates - fix validation of java.lang.Object

- resolves #7845

geoand committed:
Ensure that enums can be used in @ParameterizedTest in @QuarkusTest

Fixes: #7916

wburns committed:
Update Infinispan to 10.1.5.Final and update test classes

dwamara committed:
Placeholders section - Unnecessary lines removed

sberyozkin commented:
@lordofthejars IMHO it is a good idea to introduce `VaultBuildTimeConfig` but it should probably have an `enable` property only which enables/disables all the Vault extension.
I'm also not sure a `Health` config group is needed because the standby and perf boolean properties have gone, I've only proposed to have a group because there were 3 health related properties :-).

I wonder, is `health.enabled` property needed at all ? Vault `enable` property should be enough, if Vault is enabled (which is default) then letting the users check its status is reasonable ? This Vault `health` path can be protected if needed. CC @vsevel

Postremus committed:
Have the Mongo Clients initialized eagerly

gastaldi committed:
Add Ennovative Solutions to ADOPTERS.md

oztimpower committed:
Add quarkus-test-amazon-lambda to BOM, and remove redundant quarkus version in archetype

lordofthejars commented:
@sberyozkin As I understood, about these two properties we decided to drop until someone requests them. About the enable flag it was discussed here: https://github.com/quarkusio/quarkus/issues/7007#issuecomment-583395622

geoand committed:
Add section about how to configure the banner-started.adoc

Co-Authored-By: Guillaume Smet <guillaume.smet@gmail.com>

mkouba committed:
Qute type-safe templates - fix inconsitency in "raw" validation

- "{@java.lang.Object obj}{obj.raw}" should not result in an incorrect
expression
- follows up on #7771

manovotn committed:
Fix interceptor resolution process for multiple bindings, add automated test.

wburns committed:
Removed incorrect comment

gsmet committed:
Use quarkus-smallrye-metrics extension instead of smallrye-metrics

mkouba committed:
Support static CDI observer methods

- resolves #7938
- also split the BeanGenerator#implementCreate() method for better
readability

sberyozkin commented:
@lordofthejars OK, if it is preferred to have it optionally enabled, then it is fine. 

Re those 2 extra health properties. Since they are gone, `Health` group is not needed, however may be lets keep it if Vincent will prefer to have them configured.
What I was saying earlier, is that from the user code point's of view, typing `health(false, false)` for the mainstream case seemed too much. I'd rather just leave `health()` only and let users tweak how Vault responds with those 2 extra booleans in the `Health` group. Yes, I also typed the proposal to have `health(boolean, boolean)` :-), but if Vincent supports configuring those 2 props too then it would become a bit messy with describing what wins in `health(boolean, boolean)` vs the configuration. Not a big deal, but please think if it is reasonable...
Thanks.
Overall looks as a very useful addition IMHO

geoand committed:
Add the ability to capture the build logs in QuarkusProdModeTest

Co-Authored-By: George Gastaldi <gegastaldi@gmail.com>

geoand committed:
Replace Spring Data maven-invoker style tests with QuarkusProdModeTest

Unblocks: #7922

lordofthejars commented:
@sberyozkin Ok, tomorrow morning I'll wake up a bit earlier and I'll send the PR so you can see it and review.

iocanel committed:
chore: bump to the latest dekorate version.

sberyozkin commented:
Hey @lordofthejars look forward to it and I'm sorry for sending the confusing messages earlier :-) thanks

stuartwdouglas committed:
Remove usage of deprecated WildFlyElytronProvide

Fixes #3227

stuartwdouglas committed:
Remove unused ClassOutput

Fixes #2923

stuartwdouglas committed:
Throw exception if using blocking mailer from IO thread

This prevents a potential deadlock.

Fixes #3223

stuartwdouglas committed:
Ignore subclaases if bean is ambiguous

This fixes a whole class of bugs where adding
a subclass will prevent injection into the parent.

Fixes #3369

dependabot-preview[bot] committed:
Bump rest-assured.version from 4.2.0 to 4.3.0

Bumps `rest-assured.version` from 4.2.0 to 4.3.0.

Updates `rest-assured` from 4.2.0 to 4.3.0
- [Release notes](https://github.com/jayway/rest-assured/releases)
- [Changelog](https://github.com/rest-assured/rest-assured/blob/master/changelog.txt)
- [Commits](https://github.com/jayway/rest-assured/compare/rest-assured-4.2.0...rest-assured-4.3.0)

Updates `json-schema-validator` from 4.2.0 to 4.3.0

Signed-off-by: dependabot-preview[bot] <support@dependabot.com>

stuartwdouglas committed:
Integrate vert.x routes with security

Fixes #4544

stuartwdouglas committed:
Document extra log levels

stuartwdouglas committed:
Better error message if Quarkus version does not match

lordofthejars commented:
@sberyozkin  No worries, changes done.

aloubyansky committed:
Load wagon providers from the classloader that loaded the bootstrap classes instead of the current classloader which may already be an application Quarkus classloader

Ladicek committed:
fix NPE in QuarkusTestExtension

sberyozkin commented:
Hi @lordofthejars Thanks, IMHO it is nearly ready to go, pending an approval from @vsevel. Also CC @gsmet in case Guillaume may have some comments too. Oh yeah, can you squash please :-) ?
May be also consider dropping `ConfigItem` `name` property for all the health properties. 
Thanks

lordofthejars commented:
@sberyozkin Squashed and name property dropped down.

iocanel committed:
fix: duplicate envFrom in k8s manifests

ppalaga committed:
Fix #7964 quarkus-jdbc-mysql-deployment missing the deployment BOM

gastaldi committed:
Align jboss-parent versions in independent-projects

geoand committed:
Fix request hang issue when challenge completes exceptionally

Fixes: #7965

cescoffier committed:
reduce log level of the message printed when metrics are disabled

gsmet committed:
Some fixes in the OpenShift guide

mayankkunwar committed:
Adding Narayana depencencies to dependabot config.yml

gsmet committed:
Fix Maven minimal version

sberyozkin reviewed (approved):
@lordofthejars thanks; lets wait for the feedback from @vsevel

oztimpower committed:
Alexa SkillStreamHandler subclass check for RequestStreamHandler

geoand committed:
Remove now obsolete guide

geoand committed:
Rename kubernetes doc to be more consistent

geoand committed:
Bump jib version

geoand committed:
Fix registry setting in kubernetes doc

vsevel commented:
dear @lordofthejars & @sberyozkin,
this is some very nice work.
my main comment revolves around some exchanges we had earlier about this feature's applicability in https://github.com/quarkusio/quarkus/issues/7007#issuecomment-583395622
It is disabled by default and it is a good thing. I think we should stress in the documentation when to activate it, and/or the downsides of activating it.
People will have to realize that if they activate that feature, and vault goes rogue, the quarkus application will become not ready, will start not responding to external invocations, and most likely will trigger some monitoring alerts. If you have 200 microservices fetching secrets from vault, and vault gets down, all 200 microservices will turn red, whereas there is nothing wrong with those microservices. and it might take some time for ops to figure out that 200 incidents are just one.
To me when vault gets down, it should be red in the monitoring console, and all services depending on it should be orange to indicate that they are not running at 100% capacity.
So the vault healthcheck is really useful in situations where your microservice is dependent upon vault for 100% of its use cases (which I think is going to be rare). And still, making the service not ready is probably not going to make any difference for my clients since they are going to get a 503 instead of a 500. 
That said I am seeing other healthcheck impl for external systems (neo4j, mongo, agroal - but not for ES). So may be this is the norm?

To wrap up on this issue I would add some doc saying:
> activate this healthcheck in situations where you do not want your application to be reachable if vault is not available (for instance in situations where all your microservice's features are dependent on vault). 
> understand that if vault goes down, or switches to one of the statuses picked up by the healthcheck, all microservice replicas will become unreachable, as if they are not running.

Am I too dramatic? ;-)

other minor comments:
 - there is a bit of copy/pasting between `getBuildConfig()` and `getConfig()` in `VaultConfigSource`
 - I had already implemented the `sys/health` endpoint, but it was only used for testing purposes. so we located it into `io.quarkus.vault.test.client.dto` and used only from `TestVaultClient`, with other use cases such as `sys/init`, `sys/seal-status`, ... My version had a complete representation of the response, but is lacking the 2 optional arguments (`standbyok ` and `perfstandbyok`). it would be nice to fully implement the `sys/health` api (including the other arguments such as `activecode`, `standbycode`, ...), plus remove my old code from `TestVaultClient` and use the new code in `VaultTestExtension.initVault()`

what do you think?

gastaldi committed:
Fixed logging messages in the Gradle plugin

gastaldi committed:
Align quarkus-http version in tools-parent with Quarkus

sberyozkin commented:
Hi @vsevel Thanks a million for the feedback, I'll check in detail a bit later, lets discuss until you and @lordofthejars agree.

lordofthejars commented:
Ok, thins I am going to do:

. Resolve conflicts
. Update the sys API with the code that was developed for testing purposes
. Update docs with a warning
. Remove the Test client for doing _sys_ calls and use the new one

lordofthejars commented:
Damn it Git. Now let me check how I fix this

lordofthejars commented:
I close this PR and I open a new one with all this thing fixed. Sorry.
